### PR TITLE
feat: Configure Ruby Apiary autosynth job

### PR DESCRIPTION
**Pull request by dazuma**
Previously, these libraries were generated via an App Engine cron job (implemented in https://github.com/googleapis/discovery-artifact-manager). We are migrating this mechanism to autosynth. A synth script is already [provided](https://github.com/googleapis/google-api-ruby-client/pull/912).

The Ruby Apiary libraries are in a monorepo. Regeneration takes place via N+1 synth jobs where N is the number of API/versions. (e.g. cloudtrace/v1 and cloudtrace/v2 are generated separately.) The +1 job is a special job that deletes any clients that have been removed from the discovery list. This behavior is consistent with the current App Engine job.

Note that, currently, all clients are released into a single monopackage. This means that we *could* have configured a single synth job that regenerates all libraries, rather than N+1 jobs. We have chosen to split the synth job for two reasons: (1) we intend to split the monopackage shortly, and (2) we want separate PRs to be opened (and thus separate conventional commits created) for each generated service, so that separate line items are created in the changelog.

### CODE REVIEW

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]
Please fix the linter error.

Also, in the PR description, please describe how Ruby Apiary libraries used to be generated.

File: autosynth/multi.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def load_config(
     """
     if os.path.exists(config):
         with open(config) as fh:
-            return yaml.load(fh)["libraries"]
+            return yaml.load(fh, Loader=yaml.FullLoader)["libraries"]
```
Comment (SurferJeffAtGoogle) [inline autosynth/multi.py:275]: Why was this change necessary?

File: autosynth/providers/ruby_apiary.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            "args": ["--clean"],
+            "pr-title": "feat: Automated removal of obsolete clients",
+        }
+    )
+
+    return repositories
+
+
+def _client_dir(name: str, version: str) -> str:
+    name = re.sub(r"([A-Z]+)([A-Z][a-z])", r"\1_\2", name)
```
Comment (SurferJeffAtGoogle) [inline autosynth/providers/ruby_apiary.py:46]: Add a doc string describing what this function does.

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]

File: autosynth/multi.py (hunk 1 above)
Comment (dazuma) [reply autosynth/multi.py:275]: This was in response to your last comment to fix lint warnings. It was being flagged in the logs in the kokoro job.

File: autosynth/providers/ruby_apiary.py (hunk 2 above)
Comment (dazuma) [reply autosynth/providers/ruby_apiary.py:58]: Done

File: autosynth/multi.py (hunk 1 above)
Comment (SurferJeffAtGoogle) [reply autosynth/multi.py:275]: 😳

**Review by SurferJeffAtGoogle — APPROVED** [review]
